sound/soc/sof/intel/cnl.c

```diff
@@ -98,6 +98,9 @@ irqreturn_t cnl_ipc4_irq_thread(int irq, void *context)
 		/* This interrupt is not shared so no need to return IRQ_NONE. */
 		dev_dbg_ratelimited(sdev->dev, "nothing to do in IPC IRQ thread\n");
 
+	if (sdev->ipc->sending_delayed)
+		cnl_ipc4_send_msg(sdev, sdev->msg);
```

Comment:
if cnl_ipc4_send_msg do nothing when it is still busy and there is no more interrupt, the pending ipc msg can't be sent

Reply:
If it is busy then there _will_ be interrupt, which is the ack.
If there is no interrupt then it is a timeout as we can not send the message.